browser/android/youtube_script_injector/youtube_script_injector_tab_helper_browsertest.cc

```diff
@@ -47,8 +48,27 @@ class AndroidYouTubeScriptInjectorBrowserTest : public PlatformBrowserTest {
 
     base::FilePath test_data_dir = GetTestDataDir();
 
-    https_server_.ServeFilesFromDirectory(test_data_dir);
-    content::SetupCrossSiteRedirector(&https_server_);
+    https_server_.RegisterRequestHandler(base::BindRepeating(
```

Comment:
I don't understand what you're trying to do here, you don't need a handler to read files from disk, just set the correct directory for the https_server

Reply:
oic what you're doing, you're mapping yt requests to files, but doing it by updating a var passed as a reference is not a great way to handle this. This would be slightly better if it was bound to a method on `AndroidYouTubeScriptInjectorBrowserTest` instead of a lambda with a reference to `file_to_serve_`, but just use different v parameters and map those to the correct file. Something like `watch?v=ytcfg_mock`